app/javascript/controllers/tab_error_controller.js

```diff
@@ -24,4 +33,20 @@ export default class extends Controller {
       bootstrap.Tab.getOrCreateInstance(firstErrorTabEl).show() // eslint-disable-line no-undef
     }
   }
+
+  clearInvalidStatus(paneAndTabId) {
+    const tabElement = this.tabTargets.find(tab => tab.id === `${paneAndTabId}-tab`)
+    const paneElement = this.paneTargets.find(pane => pane.id === `${paneAndTabId}-pane`)
+    const invalidElements = paneElement.querySelectorAll('.is-invalid')
+    if (invalidElements.length > 0) {
+      invalidElements.forEach(invalidElement => {
+        if (invalidElement.tagName === 'DIV') {
+          invalidElement.remove()
+        } else {
+          invalidElement.classList.remove('is-invalid')
+        }
+      })
+      tabElement.classList.remove('is-invalid')
+    }
+  }
```

Comment:
This allows other controllers to specify which pane/tab should be cleared of invalid status due to an action taken in a controller other than this one, e.g., when a file is uploaded via dropzone or when the work type is twiddled.